compiler/rustc_middle/src/ty/visit.rs

```diff
@@ -77,30 +76,18 @@ impl<'tcx> TyCtxt<'tcx> {
 
             fn visit_region(&mut self, r: ty::Region<'tcx>) -> Self::Result {
                 match *r {
-                    ty::ReBound(debruijn, _) if debruijn < self.outer_index => {
-                        ControlFlow::Continue(())
-                    }
-                    _ => {
-                        if (self.callback)(r) {
-                            ControlFlow::Break(())
-                        } else {
-                            ControlFlow::Continue(())
-                        }
-                    }
+                    ty::ReBound(debruijn, _) if debruijn < self.outer_index => false,
+                    _ => (self.callback)(r),
                 }
             }
 
             fn visit_ty(&mut self, ty: Ty<'tcx>) -> Self::Result {
                 // We're only interested in types involving regions
-                if ty.flags().intersects(TypeFlags::HAS_FREE_REGIONS) {
-                    ty.super_visit_with(self)
-                } else {
-                    ControlFlow::Continue(())
-                }
+                ty.flags().intersects(TypeFlags::HAS_FREE_REGIONS) && ty.super_visit_with(self)
```

Comment:
If anything this is arguably harder to reason about, since the control flow is implicit here 😭

Reply:
Like, easier to read, but harder to reason about, if that makes sense